src/contract.ts

```diff
@@ -111,25 +113,35 @@ export class AbstractContract {
 
     let error = `VerifyError: ${bsi.errstr}`;
 
-    const lastStepfExec = steps[stepCounter - 1].fExec;
 
-    if(!lastStepfExec) {
-      stepCounter = AbstractContract.findLastfExec(steps, stepCounter);
-    }
 
     // some time there is no opcodes, such as when sourcemap flag is closeed. 
     if(opcodes) {
       const offset = unlockingScriptASM.trim().split(' ').length;
       // the complete script may have op_return and data, but compiled output does not have it. So we need to make sure the index is in boundary.
-      const pc = Math.min(stepCounter -  offset,  opcodes.length -1);
-  
+
+      const lastStepIndex = AbstractContract.findLastfExec(steps, stepCounter);
+
+      if(this._dataPart !== undefined && this._dataPart !== null) {
+        opcodes.push({opcode: 'OP_RETURN', file: undefined, line: undefined, endLine: undefined, column: undefined, endColumn: undefined, stack:[]});
```

Comment:
This does not look good, maybe refactor like below
```
export interface OpCode {
	file?: string;
	line?: number;
	endLine?: number;
	column?: number;
	endColumn: number;
	opcode: string;
	stack: string[];
	debugTag?: DebugModeTag;
}
```
->
```
export interface Pos {
	file?: string;
	line?: number;
	endLine?: number;
	column?: number;
	endColumn: number;
}
export interface OpCode {
	opcode: string;
	stack: string[];
         pos?: Pos;
	debugTag?: DebugModeTag;
}
```

Reply:
done

Reply:
4e6853c0d45ee3524a7c726ae3fca1a984d9d3c9